omega_client/fpg/fpg_lib.py

```diff
@@ -0,0 +1,91 @@
+# external imports
+import json, hmac, hashlib, logging, time, requests
+from requests.auth import AuthBase
+from typing import Dict
+# omega library imports
+from omega_client.messaging.common_types import AccountInfo, Order, OrderType
+
+logger = logging.getLogger(__name__)
+
+FPG_BASE_URL = 'https://fund3-staging.floating.group'
+ORDERS_URL = FPG_BASE_URL + '/v1/orders'
+
+
+class FPGAuth(AuthBase):
+    def __init__(self, api_key: str, secret_key: str):
+        self.api_key = api_key
+        self.secret_key = secret_key
+
+    def __call__(self, request):
+        timestamp = str(int(time.time()))
+        message = timestamp + self.api_key + request.path_url
+        signature = hmac.new(bytes(self.secret_key, 'latin-1'),
+                             bytes(message, 'latin-1'),
+                             hashlib.sha256).hexdigest()
+
+        request.headers.update({
+            'FPG-ACCESS-SIGN': signature,
+            'FPG-ACCESS-TIMESTAMP': timestamp,
+            'FPG-ACCESS-KEY': self.api_key,
+        })
+        return request
+
+
+def create_fpg_SOR_order(order: Order,
+                         accounts: Dict[str, AccountInfo],
+                         auth: FPGAuth):
+    """
+
+    :param order: (Order) parent Order to be spilt up amongst exchanges (
+    account_info field is discarded)
+    :param accounts: (Dict[str, AccountInfo]) dict of exchanges: AccountInfo
+    for which we will split up orders on
+    :param auth: (FPGAuth) authentication used to sign request
+    :return: List[Order] list of child Orders to be placed (empty list [] if
+    status code is not 200), status_code (int), error_message (str)
+    """
+    base, quote = order.symbol.split('/')[0], order.symbol.split('/')[1]
+    algo = 'DMA'
+    sor_body = {
+        "base": base,
+        "quote": quote,
+        "size": float(order.quantity),
+        "price": float(order.price),
+        "algo": algo,
+        "orderType": order.side.upper(),
+        "exchangeNames": [exchange.upper() for exchange in list(
+            accounts.keys())]
+    }
+    logger.info('json to be sent to fpg', extra={'sor_body': sor_body})
+    r = requests.post(ORDERS_URL, json=sor_body, auth=auth)
+    status_code, resp = r.status_code, r.json()
+    logger.info('fpg response status code', extra={'status_code': status_code})
+    logger.info('fpg response body', extra={'response': resp})
+
+    orders = []
+    error_message = ''
+    if status_code == 200:  # successful response
+        fpg_order_list = resp.get('createOrderResponse', []).get(
```

Comment:
Suggest to wrap `fpg_order_list` into a function named `def process_fpg_response(fpg_order_list)` and returns a `f3_orders_list`